packages/connectivity/src/scp-cf/destination/destination-service.spec.ts

```diff
@@ -631,6 +632,123 @@ describe('destination service', () => {
       delete process.env.no_proxy;
     });
 
+    it('does a retry if request fails wit 500 error', async () => {
+      const response = {
+        owner: {
+          SubaccountId: 'a89ea924-d9c2-4eab-84fb-3ffcaadf5d24',
+          InstanceId: null
+        },
+        destinationConfiguration: basicDestination
+      };
+      nock(destinationServiceUri)
+        .get('/destination-configuration/v1/destinations/HTTP-BASIC')
+        .reply(500)
+        .get('/destination-configuration/v1/destinations/HTTP-BASIC')
+        .reply(200, response);
+
+      const actual = await fetchDestination(destinationServiceUri, jwt, {
+        destinationName: 'HTTP-BASIC',
+        retry: true
+      });
+      expect(actual).toEqual(parseDestination(response));
+    });
+
+    it('does a no retry if request fails with 401 error', async () => {
+      const response = {
+        owner: {
+          SubaccountId: 'a89ea924-d9c2-4eab-84fb-3ffcaadf5d24',
+          InstanceId: null
+        },
+        destinationConfiguration: basicDestination
+      };
+      const mock = nock(destinationServiceUri)
+        .get('/destination-configuration/v1/destinations/HTTP-BASIC')
+        .reply(401)
+        .get('/destination-configuration/v1/destinations/HTTP-BASIC')
+        .reply(200, response);
+
+      await expect(
+        fetchDestination(destinationServiceUri, jwt, {
+          destinationName: 'HTTP-BASIC',
+          retry: true
+        })
+      ).rejects.toThrow();
+      expect(mock.isDone()).toBe(false);
+      nock.cleanAll();
+    });
+
+    it('does a retry if auth token contains errors', async () => {
+      const responseErrorInToken = {
+        owner: {
+          SubaccountId: 'a89ea924-d9c2-4eab-84fb-3ffcaadf5d24',
+          InstanceId: null
+        },
+        destinationConfiguration: oauth2SamlBearerDestination,
+        authTokens: [
+          {
+            error: 'ERROR'
+          }
+        ]
+      };
+      const responseValidToken = {
+        ...responseErrorInToken,
+        authTokens: [
+          {
+            type: 'Bearer',
+            value: 'some.token',
+            expires_in: '3600',
+            error: '',
+            http_header: {
+              key: 'Authorization',
+              value: 'Bearer some.token'
+            } as any // TODO fix the authTokens type in DestinationJson
```

Comment:
[q] Is there a ticket for this?

Reply:
None that I could find, but I am also not entirely aware of the issue. I have created a note in the backlog for now.